tests/memoryview/memoryview.pyx

```diff
@@ -261,7 +261,7 @@ def nested_struct(NestedStruct[:] mslice):
     """
     buf = mslice
     d = buf[0]
-    print d['x']['a'], d['x']['b'], d['y']['a'], d['y']['b'], d['z']
+    print d.x.a, d.x.b, d.y.a, d.y.b, d.z
```

Comment:
This is actually a pretty cool improvement. :)

Reply:
The inference is nice. But looking at it again I think I missed the point of the test: I think you could always do `memslice[0].x.a` etc. I think the test is actually look at whether a valid conversion of a Python structure exists, so I should add `cdef object buf = mslice` instead.